### PR TITLE
PP-369: Output from qstat -Bf is reportinging a negative value for waiting jobs in state_count

### PR DESCRIPTION
**Pull request by riyazhakki**
#### Issue-ID

* **[PP-369](https://pbspro.atlassian.net/browse/PP-369)**

#### Problem description
* state_count value at server was incorrect

#### Cause / Analysis
* After server gets restart job in waiting state will move to queue state and while state change happen during server restart pbsd_init_reque() increments state_count value twice for a queue job 

#### Solution description
* Please describe your changes in detail for reviewers

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [x] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
- [ ] I have added  **manual test(s) to the Jira ticket and explained why PTL is not appropriate** for this case.
- [ ] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to the Jira ticket** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: test/tests/pbs_pp369.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+# them - whether embedded or bundled with other software - under a commercial
+# license agreement.
+#
+
+from ptl.utils.pbs_testsuite import *
+
+class TestPp369(PBSTestSuite):
+	def test_job_state_count(self):
+		"""
+		test_job_state_count: Testing jobs state count value at server, when server gets restarted.
```
Comment (bhroam) [inline test/tests/pbs_pp369.py]: You're doc string is a little vague.  You're actually checking to see if jobs in the 'W' state will cause the state_count to go negative.  You might want to be a bit more precise.

File: src/server/pbsd_init.c [hunk 2]
```diff
@@ -1829,6 +1829,12 @@ pbsd_init_reque(job *pjob, int change_state)
 		/* update the state, typically to some form of QUEUED */
 		svr_evaljobstate(pjob, &newstate, &newsubstate, 1);
 		(void)svr_setjobstate(pjob, newstate, newsubstate);
+		if (newstate == JOB_STATE_QUEUED)
+			/* count for this sate is increamented
+ 			 * in svr_setjobstate() and same will 
+ 			 * again increament by svr_enquejob()
+ 			 */
```
Comment (bhroam) [inline src/server/pbsd_init.c]: Maybe reword your comment a little?  Something like: "The state count will be incremented in both svr_setjobstate() and svr_enquejo().  Decrement here so the counts do not get off."

**Comment by bhroam** [comment]
Any idea why the CLA didn't report back?  You might have to close and reopen the request to kick it again.

**Comment by riyazhakki** [comment]
Hi Bhroam,
Thanks for your review. I have addressed your comments.

**Comment by bhroam** [comment]
Thanks for making the changes.  I sign off.

**Comment by bremanandjk** [comment]
Thanks for making the changes. I sign off.

**Comment by riyazhakki** [comment]
@bremanandjk, thanks for getting my code changes and sing off.
@bhroam, Thanks for your sign off. I have squashed my commits into one, now you can take it forward.

**Comment by bhroam** [comment]
Everything appears to be in order.  I merged in the request.
